cheatsheets/DotNet_Security_Cheat_Sheet.md

```diff
@@ -297,38 +473,26 @@ DO NOT: Tell someone if the account exists on LogOn, Registration or Password re
 
 The feedback to the user should be identical whether or not the account exists, both in terms of content and behaviour: e.g. if the response takes 50% longer when the account is real then membership information can be guessed and tested.
 
-## A3 Cross Site Scripting
+### Missing function-level access control
 
-DO NOT: Trust any data the user sends you, prefer white lists (always safe) over black lists
-
-You get encoding of all HTML content with MVC3, to properly encode all content whether HTML, javascript, CSS, LDAP etc use the Microsoft AntiXSS library:
-
-`Install-Package AntiXSS`
-
-Then set in config:
+DO: Authorize users on all externally facing endpoints. The .NET framework has many ways to authorize a user, use them at method level:
 
-```xml
-<system.web>
-<httpRuntime targetFramework="4.5" 
-enableVersionHeader="false" 
-encoderType="Microsoft.Security.Application.AntiXssEncoder, AntiXssLibrary" 
-maxRequestLength="4096" />
+```csharp
+[Authorize(Roles = "Admin")]
+[HttpGet]
+public ActionResult Index(int page = 1)
 ```
 
-DO NOT: Use the `[AllowHTML]` attribute or helper class `@Html.Raw` unless you really know that the content you are writing to the browser is safe and has been escaped properly.
-
-DO: Enable a [Content Security Policy](https://developers.google.com/web/fundamentals/security/csp/), this will prevent your pages from accessing assets it should not be able to access (e.g. a malicious script):
+or better yet, at controller level:
 
-```xml
-<system.webServer>
-    <httpProtocol>
-        <customHeaders>
-            <add name="Content-Security-Policy" 
-                value="default-src 'none'; style-src 'self'; img-src 'self'; 
-                font-src 'self'; script-src 'self'" />
+```csharp
+[Authorize]
+public class UserController
 ```
 
-## A4 Insecure Direct object references
+You can also check roles in code using identity features in .net: `System.Web.Security.Roles.IsUserInRole(userName, roleName)`
+
+### Insecure Direct object references
```

Comment:
It miss the info link to the [IDOR CS](https://github.com/OWASP/CheatSheetSeries/blob/master/cheatsheets/Insecure_Direct_Object_Reference_Prevention_Cheat_Sheet.md) like for the LDAP injection